docs/onclusterlayering-quickstart.md

```diff
@@ -0,0 +1,492 @@
+    containerFile:
+    # Here is where you can set the Containerfile for your MachineConfigPool.
+    # You'll note that you need to provide an architecture. This is because this
+    # will eventually support multiarch clusters. For now, only noArch is
+    # supported.
+    - containerfileArch: noarch
+      content: |-
+        <containerfile contents>
+    # Here is where you can select an image builder type. For now, we only
+    # support the "PodImageBuilder" type that we maintain ourselves. Future
+    # integrations can / will include other build system integrations.
+    imageBuilder:
+      imageBuilderType: PodImageBuilder
+    # The Machine OS Builder needs to know what pull secret it can use to pull
+    # the base OS image.
+    baseImagePullSecret:
+      name: <secret-name>
+    # Here is where you specify the name of the push secret you use to push
+    # your newly-built image to.
+    renderedImagePushSecret:
+      name: <secret-name>
+    # Here is where you specify the image registry to push your newly-built
+    # images to.
+    renderedImagePushspec: <final image pullspec>
+  buildOutputs:
+    # Here is where you specify what image will be used on all of your nodes to
+    # pull the newly-built image.
+    currentImagePullSecret:
+      name: <secret-name>
+EOF
+
+# Write the Containerfile to a file:
+cat << EOF > Containerfile
+FROM configs AS final
+RUN rpm-ostree install tree && \
+    ostree container commit
+EOF
+
+# Finally, we'll modify our file using yq (https://github.com/mikefarah/yq).
+
+# This is the name of the secret that will be used to push the built image to
+# the image registry.
+export pushSecretName="builder-dockercfg-123"
+
+# This is the name of the secret that will be used to pull the built image from
+# the image registry onto each node.
+export pullSecretName="builder-dockercfg-123"
+
+# This is the name of the secret that will be used to pull the base OS image to
+# the build pod to be consumed during the build.
+export baseImagePullSecretName="global-pull-secret-copy"
+
+# This has the contents of the Containerfile.
+export containerfileContents="$(cat Containerfile)"
+
+# Notice that we added ":latest" onto our image registry pullspec. This tag will
+# not actually be used.
+export imageRegistryPullspec="image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-images:latest"
+
+
+yq -i e '.spec.buildInputs.baseImagePullSecret.name = strenv(baseImagePullSecretName)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildInputs.renderedImagePushSecret.name = strenv(pushSecretName)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildOutputs.currentImagePullSecret.name = strenv(pullSecretName)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildInputs.containerFile[0].content = strenv(containerfileContents)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildInputs.renderedImagePushspec = strenv(imageRegistryPullspec)' ./layered-machineosconfig.yaml
+```
```

Comment:
Do I have this right? 
Under _Write the sample MachineOSConfig to a YAML file_, the user creates the YAML file exactly as shown, with the variables (ie name: <secret-name>)? Then use the envvars and yq commands to populate those fields in the YAML?